src/App.tsx

```diff
@@ -15,37 +15,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const showList = (arr: string[]) =>
+  arr.map(good => (
+    <li key={good} data-cy="Good">
+      {good}
+    </li>
+  ));
+
+const defaultList = showList(goodsFromServer);
+
+export const App = () => {
+  const [actualList, setActualList] = useState(defaultList);
+  const [actualListArr, setActualListArr] = useState([...goodsFromServer]);
+  const [sortMethod, setSortMethod] = useState<string | null>(null);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortArray = (arr: string[], method: string): string[] => {
+    let sortedArr;
+
+    switch (method) {
+      case 'alphabetically':
+        sortedArr = [...arr].sort((a, b) => a.localeCompare(b));
+        break;
+      case 'byLength':
+        sortedArr = [...arr].sort((a, b) => a.length - b.length);
+        break;
+      default:
+        sortedArr = arr;
+    }
+
+    return sortedArr;
```

Comment:
The function 'sortArray' should be named with a verb at the beginning according to the [CODE STYLE] checklist. A more descriptive name could be 'getSortedArray' to reflect its purpose of returning a sorted array.